src/tool_parsecfg.c

```diff
@@ -57,25 +62,33 @@ int parseconfig(const char *filename, struct GlobalConfig *global)
 
 #ifndef __AMIGA__
     char *home = homedir();    /* portable homedir finder */
-    filename = CURLRC;   /* sensible default */
+    filename = CURLRC;         /* sensible default */
     if(home) {
-      if(strlen(home) < (sizeof(filebuffer) - strlen(CURLRC))) {
-        msnprintf(filebuffer, sizeof(filebuffer),
-                  "%s%s%s", home, DIR_CHAR, CURLRC);
```

Comment:
Personally, both to make the patch more reviewable *and* to make it harder for bugs to creep in, I would factor out the change from `CURLRC` to `filename` into a preparatory commit.

Reply:
Agreed - I do like to split my commits up as you have probably seen from my endeavours in curl over the years. However this is just of PoC to get feedback and interest at present.